src/librustc_mir/monomorphize/collector.rs

```diff
@@ -851,8 +851,9 @@ fn find_vtable_types_for_unsizing<'tcx>(
     }
 }
 
-fn create_fn_mono_item(instance: Instance<'_>) -> MonoItem<'_> {
+fn create_fn_mono_item<'tcx>(tcx: TyCtxt<'tcx>, mut instance: Instance<'tcx>) -> MonoItem<'tcx> {
     debug!("create_fn_mono_item(instance={})", instance);
+    instance.polymorphize(tcx);
     MonoItem::Fn(instance)
 }
```

Comment:
Hmm, this function is tiny, it might be nicer to have `MonoItem::Fn(instance.polymorphize(tcx))` at callsites instead. Although that is longer.

Reply:
I've fixed this in ac01dac.